src/Data/List/ZipList.purs

```diff
@@ -25,6 +28,24 @@ newtype ZipList a = ZipList (List a)
 runZipList :: forall a. ZipList a -> List a
 runZipList (ZipList xs) = xs
 
+-- | Transpose any structure which is both `Foldable` and `Unfoldable`,
+-- | via `ZipList`; note that `sequence` for `ZipList` is the same as
+-- | `transpose`.
+transpose :: forall f a. (Foldable f, Unfoldable f) => f (f a) -> f (f a)
```

Comment:
I thought about this, but what about name collisions? There's always straight `sequence` if you want to avoid all the other stuff, eg for performance.